teku/src/test/java/tech/pegasys/teku/cli/options/DataOptionsTest.java

```diff
@@ -67,10 +78,16 @@ public void dataStorageFrequency_shouldDefault() {
 
   @Test
   public void dataStorageFrequency_shouldAcceptNonDefaultValues() {
-    final StorageConfiguration config =
-        getTekuConfigurationFromArguments("--data-storage-archive-frequency", "1024000")
-            .storageConfiguration();
+    TekuConfiguration tekuConfiguration =
+        getTekuConfigurationFromArguments("--data-storage-archive-frequency", "1024000");
+    final StorageConfiguration config = tekuConfiguration.storageConfiguration();
     assertThat(config.getDataStorageFrequency()).isEqualTo(1024000L);
+    assertThat(
+            createConfigBuilder()
+                .storageConfiguration(b -> b.dataStorageFrequency(1024000L))
+                .build())
+        .usingRecursiveComparison()
+        .isEqualTo(tekuConfiguration);
```

Comment:
hmm, I'm not real keen on changing all the tests to work this way.  It means they aren't focussed on testing one specific thing.

Can we get the same effect of ensuring the defaults match by having a single test that creates a default `TekuConfiguration` via the config builder and one via `getTekuConfigurationFromArguments()` and checking they're the same?

Reply:
> Can we get the same effect of ensuring the defaults match by having a single test that creates a default `TekuConfiguration` via the config builder and one via `getTekuConfigurationFromArguments()` and checking they're the same?

That was my original idea, but I'd like to check that different CLI options results in the same `TekuConfiguration` because some CLI options may alter another options in the config (like e.g. `--data-path`) and I'd like to make sure these related options are modified exactly the same way as when we creating config via the `Builder`. 

I could just copy those tests as they are now into another package and revert `*OptionsTest` to their original state but does that duplication really make sense?

Reply:
Moved the default config test case from `DepositOptionsTest` to the `BeaconNodeCommandTest` to make it kind of more explicit: 35bf86b + 2ef0129

Reply:
hmm, I guess we just need to live with the extra complexity in the tests then. We'll see how it goes and if it gets frustrating for some reason we can see what other approaches could work.